content/chapitres/tech-lifecycle.adoc

```diff
@@ -0,0 +1,466 @@
+** Format de fichier `.adoc`
+** Exemple de contenu Asciidoctor :
+
+[source,asciidoctor]
+----
+= Hello, AsciiDoc!
+Doc Writer <doc@example.com>
+
+An introduction to http://asciidoc.org[AsciiDoc].
+
+== First Section
+
+* item 1
+* item 2
+
+[source,ruby]
+puts "Hello, World!"
+----
+
+== Exemple Make : Asciidoctor -> HTML
+
+* Pour convertir un fichier Asciidoctor and HTML:
++
+[source,bash]
+----
+asciidoctor index.adoc --out-file=./index.html
+----
+
+* Essayez par vous-même:
+** Ouvrez link:https://gitpod.io#https://github.com/cicd-lectures/demoapp[l'environnement dans GitPod,window=_blank]
+** Créez un fichier `index.adoc` et remplissez-le avec le contenu donné en exemple dans la slide précédente
+** Essayez la commande de conversion dans le terminal de GitPod
+** Ouvrez le fichier HTML généré dans GitPod, et affichez l'aperçu avec l'icône en haut à droite
+
+== Exemple Make : Récapépète
+
+image::make-flow-1.svg[width=800]
+
+== Exemple Make : Introduction à Makefile
+
+* link:https://www.gnu.org/software/make/[GNU Make] est une ligne de commande,
+* qui lit un fichier `Makefile` pour exécuter des tâches.
+* Chaque tâche (ou "règle") est décrite par une "cible":
+* Format d'une "cible" make :
++
+[source,makefile]
+----
+cible: dependance
+	commandes
+----
+* On appelle la commande `make` avec une ou plusieurs cibles en argument :
++
+[source,bash]
+----
+make clean build
+----
+
+== Exemple Make : Exemple de Makefile
+
+[source,makefile]
+----
+# Fabrique le fichier "hello" (binaire) à partir des fichier "hello.o" et "main.o"
+hello: hello.o main.o
+	gcc -o hello hello.o main.o
+
+# Fabrique le fichier "hello.o" à partir du code source "hello.c"
+hello.o: hello.c
+	gcc -o hello.o -c hello.c
+
+# Fabrique le fichier "main.o" à partir du code source "main.c"
+main.o: main.c
+	gcc -o main.o -c main.c
+----
+
+[source,bash]
+----
+make hello # Appelle implicitement "make hello.o" et "make main.o"
+## équivalent à "make hello.o main.o hello"
+----
+
+== Exemple Make : Makefile - Build
+
+* *But :* on veut générer le HTML en appelant la commande `make index.html`
+* Toujours dans link:https://gitpod.io#https://github.com/cicd-lectures/demoapp[l'environnement GitPod,window=_blank],
+supprimez le fichier `index.html` si vous l'avez déjà
+* Créez un fichier `Makefile` qui contient une cible `index.html` et qui appelle la commande `asciidoctor`:
++
+[source,Makefile]
+----
+index.html:
+  asciidoctor index.adoc --out-file=$(CURDIR)/index.html
+  # $(CURDIR) est une variable Make désignant le chemin vers le dossier contenant le Makefile
+----
+
+* Essayez avec la commande `make index.html`
+
+[.notes]
+--
+=> Le fichier index.adoc doit être présent...
+--
+
+== Exercice Make : Makefile - Package
+
+* *But :* on veut générer le livrable en appelant la commande `make example-make.tar.gz`
+* On utilisera la commande `tar xzf ./example-make.tar.gz ./index.html`
```

Comment:
```suggestion
* On utilisera la commande `tar czf $(CURDIR)/example-make.tar.gz $(CURDIR)/index.html`
```

As `CURDIR` has been already introduced ?
Also, aren't we creating an archive here :thinking:

Reply:
Double 👍 !

Reply:
Good catch for the archive creation!

You're right about the `CURDIR` (which is in the solution), but the goal is to ensure that it is understood: the "exercise" intent does not dive into this detail :)

Reply:
```suggestion
* On utilisera la commande `tar czf ./example-make.tar.gz ./index.html`
```